### PR TITLE
Annotation dragging

### PR DESCRIPTION
**Pull request by mdtusz**
Replaces https://github.com/plotly/plotly.js/pull/502.

Fixes the drag interaction on annotation text and the arrow.

### CODE REVIEW

**Comment by etpinard** [comment]
Looks good!

@coopy would you mind pulling down this branch and check if indeed it solves the issue you pointed out [here](https://github.com/plotly/plotly.js/pull/502#issuecomment-216699215) ?

**Comment by coopy** [comment]
It works like ✨  magic! ✨ 
Thank you, @mdtusz.

l9fqhzkvso

**Comment by etpinard** [comment]
:dancer:

File: test/jasmine/tests/config_test.js [hunk 1]
```diff
@@ -78,6 +81,24 @@ describe('config argument', function() {
...
+
+            mouseEvent('mousedown', elX, elY);
+            mouseEvent('mousemove', elX - 20, elY + 20);
+
+            var movedBox = el.getBoundingClientRect();
+
+            expect(movedBox.left).toBe(elBox.left - 20);
+            expect(movedBox.top).toBe(elBox.top + 20);
+
+            mouseEvent('mouseup', elX - 20, elY + 20);
```
Comment (coopy) [inline test/jasmine/tests/config_test.js]: Double ➕  good test!

**Pull request merged**